spec/assertions_spec.lua

```diff
@@ -45,6 +45,18 @@ describe("Test Assertions", function()
     assert.is_not.same(nil, "a string")
   end)
 
+  it("Checks same() assertion to handle recursive tables", function()
+    local t1 = { k1 = 1, k2 = 2, k3 = t1 }
+    local t2 = { k1 = 1, k2 = 2, k3 = t2 }
+    local t3 = { k1 = 1, k2 = 2, k3 = { k1 = 1, k2 = 2, k3 = t2 } }
+    t1.k3 = t1
+    t2.k3 = t2
+
+    assert.same(t1, t2)
```

Comment:
Wrestled with this a bit, but it makes sense; will just need to add documentation that self-referencing table values are treated as the same, even if the actual values are different.